processing/src/test/java/io/druid/query/groupby/GroupByQueryRunnerTest.java

```diff
@@ -1651,6 +1651,279 @@ public void testGroupByWithOrderLimit4()
   }
 
   @Test
+  public void testGroupByWithOrderOnHyperUnique()
+  {
+    GroupByQuery query = new GroupByQuery.Builder()
+        .setDataSource(QueryRunnerTestHelper.dataSource)
+        .setGranularity(QueryRunnerTestHelper.allGran)
+        .setDimensions(
+            Arrays.<DimensionSpec>asList(
+                new DefaultDimensionSpec(
+                    QueryRunnerTestHelper.marketDimension,
+                    QueryRunnerTestHelper.marketDimension
+                )
+            )
+        )
+        .setInterval(QueryRunnerTestHelper.fullOnInterval)
+        .setLimitSpec(
+            new DefaultLimitSpec(
+                Lists.newArrayList(
+                    new OrderByColumnSpec(
+                        QueryRunnerTestHelper.uniqueMetric,
+                        OrderByColumnSpec.Direction.DESCENDING
+                    )
+                ), 3
+            )
+        )
+        .setAggregatorSpecs(
+            Lists.<AggregatorFactory>newArrayList(
+                QueryRunnerTestHelper.qualityUniques
+            )
+        )
+        .setPostAggregatorSpecs(
+            Lists.<PostAggregator>newArrayList(
+                new HyperUniqueFinalizingPostAggregator(
+                    QueryRunnerTestHelper.hyperUniqueFinalizingPostAggMetric,
+                    QueryRunnerTestHelper.uniqueMetric
+                )
+            )
+        )
+        .build();
+
+    List<Row> expectedResults = Arrays.asList(
+        GroupByQueryRunnerTestHelper.createExpectedRow(
+            "1970-01-01T00:00:00.000Z",
+            "market",
+            "spot",
+            QueryRunnerTestHelper.uniqueMetric,
+            QueryRunnerTestHelper.UNIQUES_9,
+            QueryRunnerTestHelper.hyperUniqueFinalizingPostAggMetric,
+            QueryRunnerTestHelper.UNIQUES_9
+        ),
+        GroupByQueryRunnerTestHelper.createExpectedRow(
+            "1970-01-01T00:00:00.000Z",
+            "market",
+            "upfront",
+            QueryRunnerTestHelper.uniqueMetric,
+            QueryRunnerTestHelper.UNIQUES_2,
+            QueryRunnerTestHelper.hyperUniqueFinalizingPostAggMetric,
+            QueryRunnerTestHelper.UNIQUES_2
+        ),
+        GroupByQueryRunnerTestHelper.createExpectedRow(
+            "1970-01-01T00:00:00.000Z",
+            "market",
+            "total_market",
+            QueryRunnerTestHelper.uniqueMetric,
+            QueryRunnerTestHelper.UNIQUES_2,
+            QueryRunnerTestHelper.hyperUniqueFinalizingPostAggMetric,
+            QueryRunnerTestHelper.UNIQUES_2
+        )
+    );
+
+    Iterable<Row> results = GroupByQueryRunnerTestHelper.runQuery(factory, runner, query);
+    TestHelper.assertExpectedObjects(expectedResults, results, "order-limit");
+  }
+
+  @Test
+  @Ignore // Doesn't actually work: HavingSpecMetricComparator throws Unknown type[class io.druid.query.aggregation.hyperloglog.HLLCV1]
```

Comment:
depends on when this is slated for. If it is needed for 0.9.0 then an issue is fine, but if it is for 0.9.1 then its worth taking a bit of time to get it ironed out a bit.